cdap-watchdog/src/main/java/co/cask/cdap/metrics/collect/MessagingMetricsCollectionService.java

```diff
@@ -91,8 +94,10 @@ protected void publish(Iterator<MetricValues> metrics) throws Exception {
       recordWriter.encode(metricValues, encoder);
       // Calculate the topic number with the hashcode of MetricValues' tags and store the encoded payload in the
       // corresponding list of the topic number
-      topicPayloads.get(Math.abs(metricValues.getTags().hashCode() % size))
-        .addPayload(encoderOutputStream.toByteArray());
+      int topicNumber = Math.abs(metricValues.getTags().hashCode() % size);
+      // this log is useful in knowing to which topic the program's metrics are emitted to
+      TOPIC_LOG.debug("Metrics is emitted to topic {}", topicPayloads.get(topicNumber).topicId.getTopic());
```

Comment:
since the hash is based on the tags and tags may continue tag which is finer than the program, (mr-task or flowlet) for example. so its not guaranteed to be at program level. so i have removed this log for now.